services/src/main/java/org/keycloak/protocol/oid4vc/issuance/mappers/OID4VCGeneratedIdMapper.java

```diff
@@ -65,29 +68,33 @@ protected List<ProviderConfigProperty> getIndividualConfigProperties() {
     @Override
     public boolean includeInMetadata() {
         return Optional.ofNullable(mapperModel.getConfig().get(CredentialScopeModel.INCLUDE_IN_METADATA))
-                       .map(Boolean::parseBoolean)
-                       .orElse(false);
+                .map(Boolean::parseBoolean)
+                .orElse(false);
     }
 
     @Override
     public List<String> getMetadataAttributePath() {
         String property = Optional.ofNullable(mapperModel.getConfig())
-                                  .map(config -> config.get(CLAIM_NAME))
-                                  .orElse(SUBJECT_PROPERTY_CONFIG_KEY_DEFAULT);
+                .map(config -> config.get(CLAIM_NAME))
+                .orElse(SUBJECT_PROPERTY_CONFIG_KEY_DEFAULT);
         return ListUtils.union(getAttributePrefix(), List.of(property));
     }
 
-    public void setClaimsForCredential(VerifiableCredential verifiableCredential,
-                                       UserSessionModel userSessionModel) {
-        // nothing to do for the mapper.
+    public void setClaimsForCredential(VerifiableCredential verifiableCredential, UserSessionModel userSessionModel) {
+        // Assign a generated ID
+        URI vcId = URI.create(String.format("urn:uuid:%s", UUID.randomUUID()));
+        List<String> attributePath = getMetadataAttributePath();
+        String propertyName = attributePath.get(attributePath.size() - 1);
+        if (CLAIM_NAME_VC_ID.equals(propertyName)) {
+            verifiableCredential.setId(vcId);
+        } else {
```

Comment:
I actually thought we should set the id of the user instead of generating a random UUID in this case?